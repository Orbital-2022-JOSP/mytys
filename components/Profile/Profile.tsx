import { GenericContainer } from '../GenericContainer/GenericContainer';

type ProfileProps = {
    username?: string;
    problemsSolved?: number;
    percentageSolved?: number;
}

export const Profile: React.FC<ProfileProps> = ({ username, problemsSolved, percentageSolved }) => {
    return (
        <GenericContainer>
            <div className="grid grid-cols-12 gap-4">
                <div className="col-span-12 lg:col-span-5 xl-col-span-4 2xl-col-span-3">
                    <div className="flex flex-col mb-16 sm:text-center sm:mb-0 border border-gray-100 shadow-xl rounded-xl p-5">
                        <img
                            className="object-cover w-56 h-56 rounded-full shadow-lg mx-auto mb-10"
                            src="https://images.pexels.com/photos/927022/pexels-photo-927022.jpeg?auto=compress&amp;cs=tinysrgb&amp;dpr=3&amp;h=750&amp;w=1260"
                            alt=""
                        />
                        <h2 className="mb-5 font-sans text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl sm:leading-none md:mb-6 group">
                            {username ? username : "User"}
                        </h2>
                        {/* <p className="text-base text-gray-700 md:text-lg">
                            Sed ut perspiciatis unde omnis iste natus error sit voluptatem
                            accusantium doloremque rem aperiam, eaque ipsa quae.
                        </p> */}
                    </div>
                </div>
                <div className="col-span-12 lg:col-span-7 xl-col-span-8 2xl-col-span-9">
                    <div className="flex flex-col mb-16 sm:text-center sm:mb-0 border border-gray-100 shadow-xl rounded-xl p-5">
                        <p>Solved Problems</p>
                        <div className="grid grid-cols-12 gap-4 mt-5">
                            <div className="col-span-4">
                                <p className="text-xl font-medium">{problemsSolved ? problemsSolved : 32}</p>
                                <p className="text-sm">Problems</p>
                            </div>
                            <div className="col-span-8">
                                <div className="flex justify-between mb-1">
                                    {/* <span className="text-base font-medium text-blue-700">Flowbite</span> */}
                                    <span className="text-sm font-medium text-blue-700">{percentageSolved ? percentageSolved : 45}%</span>
                                </div>
                                <div className="w-full bg-gray-200 rounded-full h-2.5">
                                    <div className="bg-blue-600 h-2.5 rounded-full" style={{ width: percentageSolved ? percentageSolved : 45 + "%" }}></div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </GenericContainer>
    )
}