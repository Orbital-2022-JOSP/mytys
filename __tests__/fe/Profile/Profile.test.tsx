import { render } from '@testing-library/react';
import { Profile } from '../../../components/Profile/Profile';

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {
        expires: new Date(Date.now() + 2 * 86400).toISOString(),
        user: {
            id: "123",
            userRole: "student",
            name: "John",
            email: "fake@email.com"
        }
    };
    return {
        __esModule: true,
        ...originalModule,
        useSession: jest.fn(() => {
            return { data: mockSession, status: 'authenticated' }
        }),
    };
});

describe("The Profile Component", () => {

    it('Should render correctly',
        async () => {
            const { container } = render(<Profile />);

            expect(container).toMatchSnapshot()
        })
})
