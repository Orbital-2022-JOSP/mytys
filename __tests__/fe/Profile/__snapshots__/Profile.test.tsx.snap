// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Profile Component Should render correctly 1`] = `
<div>
  <div
    class="px-4 py-16 mx-auto sm:max-w-xl md:max-w-full lg:max-w-screen-xl sm:min-w-xl md:min-w-full lg:min-w-screen-lg xl:min-w-screen-xl md:px-24 lg:px-8 lg:py-20"
  >
    <div
      class="grid grid-cols-12 gap-4"
    >
      <div
        class="col-span-12 lg:col-span-5 xl-col-span-4 2xl-col-span-3"
      >
        <div
          class="flex flex-col mb-16 sm:text-center sm:mb-0 border border-gray-100 shadow-xl rounded-xl p-5"
        >
          <img
            alt=""
            class="object-cover w-56 h-56 rounded-full shadow-lg mx-auto mb-10"
            src="https://images.pexels.com/photos/927022/pexels-photo-927022.jpeg?auto=compress&cs=tinysrgb&dpr=3&h=750&w=1260"
          />
          <h2
            class="mb-5 font-sans text-3xl font-bold tracking-tight text-gray-900 sm:text-4xl sm:leading-none md:mb-6 group"
          >
            User
          </h2>
        </div>
      </div>
      <div
        class="col-span-12 lg:col-span-7 xl-col-span-8 2xl-col-span-9"
      >
        <div
          class="flex flex-col mb-16 sm:text-center sm:mb-0 border border-gray-100 shadow-xl rounded-xl p-5"
        >
          <p>
            Solved Problems
          </p>
          <div
            class="grid grid-cols-12 gap-4 mt-5"
          >
            <div
              class="col-span-4"
            >
              <p
                class="text-xl font-medium"
              >
                32
              </p>
              <p
                class="text-sm"
              >
                Problems
              </p>
            </div>
            <div
              class="col-span-8"
            >
              <div
                class="flex justify-between mb-1"
              >
                <span
                  class="text-sm font-medium text-blue-700"
                >
                  45
                  %
                </span>
              </div>
              <div
                class="w-full bg-gray-200 rounded-full h-2.5"
              >
                <div
                  class="bg-blue-600 h-2.5 rounded-full"
                  style="width: 45%;"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
